_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+I'd like cover a lot of the common questions we get asked about from the perspective of native developers:
+
+-   What is React Native?
+-   How do you use React Native?
+-   When is React Native a good technology choice?
+
+This article covers an awful lot, so free up at least 30 minutes, make a tea and then come back to this on your computer. It's worth your time if you're interested in all the hype around React Native.
+
+<!-- more -->
+
+<div class="mobile-only">
+<p>
+  <strong>Before you get started</strong>, it looks like you're using a really small screen, this post is built for larger screens and having a terminal around will make it much easier to understand the ideas inside the post. <strong>So if you can, please switch to a different device.</strong> You will be missing sections otherwise.
+</p>
+</div>
+
+At the highest level, React Native is a way to write React apps that run as native programs. You write your app's code in JavaScript, and React Native bridges that code with native UIView elements. React Native has two stated aims:
+
+-   Learn Once, Write Anywhere.
+-   Make a native developer experience as fast as the web developer's.
+
+**"Learn Once, Write Anywhere"** is a play on Java's _"Write Once, Run Anywhere"_ - something that has not worked well for user-interface heavy mobile clients. The idea of running the same code everywhere encourages platform-less APIs which water down the positives of each platform.
+
+**"Learn once"** in this context means that you can re-use the same ideas and tools across many platforms. You don't lose your ability to write the same user experiences as you can with native code, but you can re-use your existing skills across different platforms. That is the **"Write Anywhere"** aspect.
+
+React Native makes it feasible to share a lot of code between iOS, Android and Web. It is not a panacea for making a cross-platform app though, cross-platform is not an _explicit_ goal of the project. The project moves towards making the best per-platform apps.
+
+To make a developer experience **as fast as the web developer's** you need to really reflect on how slow native development is. Even before Swift double/triple/quadrupled the time to compile an app, a change in one part of the app required a full restart of the simulator, and for the developer to get back into the same position to see the changes. As a web developer you would just refresh the browser. For example with the simplest Xcode iOS app template, I made a single line change and did an incremental rebuild and it took 9 seconds to get me back into my app with the new change, on a 2015 MacBook Pro. 9 seconds per change leaves a lot of time to lose focus, and discourages playfulness.
```

Comment:
I would remove the quip about Swift compile times, your argument is strong enough without it and it might alienate readers.